source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -297,13 +313,34 @@ Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
   return Http::Code::OK;
 }
 
-void HystrixSink::flush(Stats::Source&) {
+void HystrixSink::flush(Stats::Source& source) {
   if (callbacks_list_.empty()) {
     return;
   }
   incCounter();
   std::stringstream ss;
   Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+
+  // Save a map of the relevant histograms per cluster in a convenient format.
+  std::unordered_map<std::string, QuantileLatencyMap> time_histograms;
+  for (const Stats::ParentHistogramSharedPtr histogram : source.cachedHistograms()) {
+    // histogram->name() on clusters of the format "cluster.cluster_name.histogram_name"
+    // i.e. "cluster.service1.upstream_rq_time".
+    const std::vector<std::string> split_name = absl::StrSplit(histogram->name(), '.');
+    if (split_name.size() >= 2 && split_name[0] == "cluster" &&
+        split_name[2] == "upstream_rq_time") {
+      QuantileLatencyMap& hist_map = time_histograms[split_name[1]];
```

Comment:
Don't we only expect to do this emplacement once per flush?  What happens if you get multiple histograms that happen to satisfy your format checks and produce the same cluster name on the same flush (because of the dynamic nature of stats strings, things like this are always possible :) )?  Won't the second overwrite the first?  How do you want to handle that case?  

If you want to do something other than overwrite, this case can be checked pretty easily by using `time_histograms.emplace()` and checking the return value.  See http://www.cplusplus.com/reference/unordered_map/unordered_map/emplace/.

Reply:
I think there's no problem with overwriting. We want to show latency information in dashboard, it doesn't have to be exact data. If there is an extreme value (very high latency) it could be either a temporary situation that has already recovered, or a consistent problem the user wants to find and fix. In the case it was temporary, if we missed it because of the overwrite, it doesn't matter. If it is consistent, user will see it on next flush.

Reply:
That makes sense.  I think overwriting is fine.  However, it seems like the code assumes that each cluster should only have one histogram that matches this filter (a pretty good assumption IMO).  If there's another histogram that somehow gets inside the if statement, it might be worthwhile to log an error because it means that something is probably broken here.  WDYT?

Reply:
no problem, I can log in both cases, can you point me to a similar example?

Reply:
assert here too?

Reply:
Yep!